stdlib/2/typing.pyi

```diff
@@ -82,6 +82,11 @@ class SupportsComplex(Protocol, metaclass=ABCMeta):
     @abstractmethod
     def __complex__(self) -> complex: ...
 
+@runtime
+class SupportsIndex(Protocol, metaclass=ABCMeta):
```

Comment:
This doesn't exist at runtime (though perhaps it should). Feel free to suggest adding it at github.com/python/typing, but until then we should just define a private `_SupportsIndex` protocol in builtins.